src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -1000,5 +1015,154 @@ public void ReadArchive_WithUnexpectedZip64ExtraFieldSizeUncompressedSizeIn32Bit
             // comment length
             0x00, 0x00
         };
+
+        private static readonly byte[] s_zip64WithBigStartDiskNumber =
```

Comment:
Rather than duplicate this whole thing what about just flipping the byte you need, within the test?

Reply:
Good point, i'll try. But I changed quite a few bytes in different parts([https://www.diffchecker.com/YK5V7XvS](https://www.diffchecker.com/YK5V7XvS)), so any advice would be helpful for me how to do it better

Reply:
I almost wonder then whether it would make sense to extract that "proto zip" data into a mock with a method like SetDiskNumber. That would make it cleaner to use and easier to maintain and extend.

Reply:
If I understand your idea correctly. We take s_slightlyIncorrectZip64 as a basis and create extensions for it such as SetStartDiskNumber? (and smth like SetCustomHeaderField in future)

Reply:
yes -- I mean, I created that for a particular test and I see it's getting used in another place, now you needed it here and had to make larger edits as you say. But those tests shouldn't need to know the zip format.

So I wonder whether that suggests it would be helpful to wrap the array in a class that exposes it and that class can take care of any modifications a test needs. For now if nothing else you could just paste your new array in there, alongside the other one, but it would be a start.

If you don't want to do that here, I think what you have is fine. (I'll let area owners sign off though)

Reply:
To be honest, I don't have an idea for a good implementation of this right now. It is complicated by the fact that different tests use different initialization methods (zip files in ZipTestData, byte arrays through MemberData or static byte arrays) 

In my vision, it would look something like this:
```
var zipFile = FluentZipFileGenerator.CreateBuilder()
		.GetDefaultZipTemplate()
		.SetZip64ExtraFieldSize(args)
		.SetStartDiskNumber(args)
		.GenerateZipFile();
```

In this PR I can move the archive to ZipTestData/StrangeZipFiles folder.

Reply:
Maybe we should tag someone else?